catalogue/webapp/components/SearchFilters/SearchFilters.Desktop.DynamicFilters.tsx

```diff
@@ -36,17 +36,40 @@ const DynamicFilterArray = ({
     }
   };
 
-  const renderDynamicFilter = (f: Filter, i: number, arr: Filter[]) => {
+  const renderDynamicFilter = (f: Filter, i: number) => {
+    const isHidden = hasCalculatedFilters
+      ? !dynamicFilters.map(f => f.id).includes(f.id)
+      : false;
+
     // We need to have the excluded filters still in the form so their values gets retained in the URL
     // when more filtering is done (e.g. partOf.title)
-    // Only checkbox types are excluded at the moment
-    return f.excludeFromMoreFilters && f.type === 'checkbox' ? (
+    return isHidden || f.excludeFromMoreFilters ? (
       <div className="is-hidden">
-        <CheckboxFilter
-          {...(!showMoreFiltersModal && { form: searchFormId })}
-          f={f}
-          changeHandler={changeHandler}
-        />
+        {f.type === 'checkbox' && (
+          <CheckboxFilter
+            {...(!showMoreFiltersModal && { form: searchFormId })}
+            f={f}
+            changeHandler={changeHandler}
+          />
+        )}
+        {f.type === 'dateRange' && (
+          <DesktopDateRangeFilter
+            {...(!showMoreFiltersModal && { form: searchFormId })}
+            f={f}
+            changeHandler={changeHandler}
+            hasNoOptions={hasNoResults && !(f.from.value || f.to.value)}
+          />
+        )}
+
+        {f.type === 'color' && (
```

Comment:
Date Range and Color are both the second filter in their respective pages, so I could change this to only account for Checkbox filters as it becomes the mobile menu before getting too small for them. Up to us, I felt this would future proof it.